### ISSUE TITLE
Alien Hominid Crashing

### ISSUE THREAD

**Issue by ArcaneArcanine** [emukidid/swiss-gc#410]
Everytime it hits actual gameplay (story & mini-games) the game freezes and the audio glitches out terribly.

**Comment by Abro61209** [comment]
Can confirm this happens to me too. It most likely has something to do with swiss.

**Comment by Abro61209** [comment]
@ArcaneArcanine A new version of swiss was released 5 days ago, update your swiss. Alien hominid is now working flawlessly for me. Also using a 128gb fat32 PNY sd card with allocation size at 32k. Don't know if it matters but just putting it out there.

Keep up the great work, Extrems.

**Comment by Extrems** [comment]
Uh, I'm pretty sure it's still broken.

**Comment by Abro61209** [comment]
@Extrems Have you tried it? I originally had a micro SD in a SD adapter for my hombrewed cube that was formatted as ExFAT. I switched to the SD card I listed above and made it FAT32 to have compatibility with emulators and homebrew apps since they simply do not work on the ExFAT file system for obvious reasons.  Alien hominid would behave like ArcaneArcanine stated and freeze about a minute into gameplay. Even after the switch to this fat32 card it still didn't work. Also, when I made the switch to my FAT32 card, my metroid prime 1 kept giving me a black screen after the logos at the beginning, which didn't happen on my ExFAT card. It would work like normal and I even played through the whole game. Even tried a different ISO.

SO long story short I came to see if you had released any updates recently, installed your latest swiss revision, and now prime 1 is working again as well as Alien Hominid. I was able to play about 10 minutes of Alien Hominid and it didn't freeze.

Hopefully @ArcaneArcanine can give it a try too and if it doesn't work for them then I really have no idea why Alien Hominid is working for me but I'm not going to complain.

**Comment by ArcaneArcanine** [comment]
Yeah mine still freezes at the very beginning of the game but the audio isn't garbled anymore. I dont remember what I have my SD card formatted too I'll have to mess with that. Also I noticed my GBA emulator works better on my SD to memory card adapter over my  SD2SP2 so I'm gonna see if putting Alien Homind on my regular SD card will yield different results.

**Comment by chemaperez** [comment]
Is anyone looking into this? The game always freezes, is there any way to get it running?

**Comment by Extrems** [comment]
We still have no idea of the exact cause.

**Comment by Extrems** [comment]
Nothing changed since you opened this issue.

**Comment by Jaffy420** [comment]
Confirmed, freezing

**Comment by theretrokid123** [comment]
Yeah this issue is still happening :(

**Comment by Chuckles1001** [comment]
Sad  times, this would be a game that would be great to get working :)

**Comment by TwilightSlick** [comment]
Does this issue occur on GC Loader? I understand that the GC Loader does audio streaming emulation internally and not through Swiss's hypervisor, correct? So this issue wouldn't occur on that device?

If I am wrong, you might want to force clean boot on this title, making it so only a GC Loader or original disc can load it. I don't own a GC Loader (yet) but just want to be sure.

**Comment by Extrems** [comment]
It doesn't occur with the GC Loader or disc drive, clean boot or not.

**Comment by ChrisPVille** [comment]
This game is kinda obnoxious in its use of the audio interface commands. I had to do a bit of patching on the flippydrive to improve the command handling since it's constantly doubling up audio setup commands. When a track is queued up it will send:

1. Normal track setup, addr, len
2. Immediately followed by all-zeroes command to disable looping
3. Usual spam of audio status to software re-configure audio on completion

If the status bit isn't immediately set, all hell breaks loose.  Even when audio isn't playing, on some screens like the first menu, it continues to spam pairs of:

1. Zero length audio setup with valid address
2. Zero length audio setup with zero address

I have no idea if it's legal, but the drive seems to tolerate it. The only maybe relevant thing I see is [here](https://github.com/emukidid/swiss-gc/blob/7985e461f0c949379af4c5ab8ce977933d6157a0/cube/patches/base/emulator.c#L722). I think the condition for stopping the track is just length==0.  That double setup while-stopped thing the game does seems like it would cause issues with the logic.  If if would be helpful, I have full DI transaction logs of Alien Hominid running on hardware.

**Comment by Extrems** [comment]
That was taken from [Dolphin](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/HW/DVD/DVDInterface.cpp#L988), which was hardware-tested on Wii. I guess I should double-check their work?

**Comment by Extrems** [comment]
I found some inaccuracies, but they don't seem to be related as it's hanging in the middle of a track.

**Comment by Extrems** [comment]
Partially fixed in 2936e0d (r1640) for some devices.

**Comment by AntoniFloris** [comment]
hey guys  i know this issue is old but im having problem even opening the game on swiss it just appears failed to read fst anyone know a fix?

**Comment by Sneaky-Pete-Littlemin** [comment]
> oh i see i already fixed it myself

Pics or it didn't happen! Ide much love to have this game work over sd2sp2.
Any updates on this?
